classes/Kohana/Jam/Field/Integer.php

```diff
@@ -32,5 +32,36 @@ public function set(Jam_Validated $model, $value, $is_changed)
 
 		return $value;
 	}
+	
+	/**
+	 * Like Jam_Field::_default, but it won't convert 0 and 0.0 to empty_value
+	 *
+	 * @param  Jam_Validated $model
+	 * @param  mixed $value
+	 * @return array 1st element the converted value; 2nd element a boolean
+	 * indicating if the value should not be processed further
+	 */
+	protected function _default(Jam_Validated $model, $value)
+	{
+		$return = FALSE;
+
+		$value = $this->run_filters($model, $value);
+
+		// Convert empty values to NULL, if needed
+		if ($this->convert_empty AND empty($value) AND $value !== 0 AND $value !== 0.0 AND $value !== "0" AND $value !== "0.0")
```

Comment:
The issue I'm experiencing is that the field I have can also accept NULL,
so I need to be able to use the standard empty_value and in addition be
able to set 0 as a value.